src/game_events/handlers.hpp

```diff
@@ -29,17 +29,33 @@
 
 class game_data;
 class game_lua_kernel;
+class variable_set;
 
 namespace game_events
 {
 struct queued_event;
+/** Represents a single filter condition on an event. */
+struct event_filter {
+	/** Runs the filter and returns whether it passes on the given event. */
+	virtual bool operator()(const queued_event& event_info) const = 0;
+	/** Serializes the filter into a config, if possible. */
+	virtual void serialize(config& cfg) const;
+	/** Returns true if it is possible to serialize the filter into a config. */
+	virtual bool can_serialize() const;
+	virtual ~event_filter() = default;
+	event_filter() = default;
+private:
+	event_filter(const event_filter&) = delete;
+	event_filter& operator=(const event_filter&) = delete;
+};
 
 class event_handler
```

Comment:
Doxygen comments on things that have been newly added would be good.

Reply:
The documentation still seems to be missing.

Reply:
Although docs were added to `event_filter`, they're still missing from the same file's `event_handler`. While that existed before, most of its methods are new additions.